### PR TITLE
Added annotation typing to resnet

### PR DESCRIPTION
**Pull request by frgfm**
Hi there!

As per #2025, annotation typing are welcome in torchvision. So, this PR focuses on `torchvision.models.resnet` this PR!

Any feedback is welcome!

### CODE REVIEW

**Review by Sifat1998 — APPROVED** [review]

File: torchvision/models/resnet.py [hunk 1]
```diff
@@ -216,11 +245,12 @@ def _forward_impl(self, x):
...
         return x
 
-    def forward(self, x):
+    def forward(self, x: Tensor) -> Tensor:
         return self._forward_impl(x)
 
 
-def _resnet(arch, block, layers, pretrained, progress, **kwargs):
+def _resnet(arch: str, block: Type[Union[BasicBlock, Bottleneck]], layers: List[int], pretrained: bool, progress: bool,
+            **kwargs: Any) -> ResNet:
```
Comment (pmeier) [inline torchvision/models/resnet.py:253]: Could you format this with `black` to make it more legible?

**Review by pmeier — COMMENTED** [review]
Minor nitpick plus incremental changes from the other PRs. Otherwise LGTM.

File: torchvision/models/resnet.py (hunk 1 above)
Comment (frgfm) [reply torchvision/models/resnet.py]: Fixed!

**Review by fmassa — APPROVED** [review]
Thanks!

**Pull request merged**